### PR TITLE
Write and show lesson descriptions

### PR DESCRIPTION
**Pull request by bethanyaconnor**
`student_description` and `teacher_description` are in `scripts.en.yml` for older courses. This adds a pathway for saving those fields in `scripts.en.yml` and then fetching them when displaying the script overview.

Teacher:
Screen Shot 2021-02-23 at 10 19 53 AM

Student:
Screen Shot 2021-02-23 at 10 20 00 AM

I also tested saving old courses and the descriptions do not change.

Updated `scripts.en.yml` using 
```
lessons = Script.where("name like '%-2021'").select{|s| s.is_migrated?}.map(&:lessons).flatten
lessons.each {|l| Script.merge_and_write_i18n(l.i18n_hash, l.name) }
```

## Testing story

manual testing, updated unit tests

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/lesson.rb [hunk 1]
```diff
@@ -229,6 +229,11 @@ def summarize(include_bonus_levels = false, for_edit: false)
     lesson_summary = Rails.cache.fetch("#{cache_key}/lesson_summary/#{I18n.locale}/#{include_bonus_levels}") do
       cached_levels = include_bonus_levels ? cached_script_levels : cached_script_levels.reject(&:bonus)
 
+      description_student = I18n.t("data.script.name.#{script.name}.lessons.#{key}.description_student", default: '')
```
Comment (Hamms) [inline dashboard/app/models/lesson.rb]: We prefer

```suggestion
      description_student = I18n.t(
        'description_student',
        scope: [:data, :script, :name, script.name, :lessons, key],
        smart: true,
        default: ''
      )
```

for strings whose scopes include dynamic content (script name, lesson key) for which it's possible to include the `.` character

File: dashboard/app/models/lesson.rb [hunk 2]
```diff
@@ -229,6 +229,11 @@ def summarize(include_bonus_levels = false, for_edit: false)
...
+      description_student = render_codespan_only_markdown(description_student) unless script.is_migrated?
```
Comment (Hamms) [inline dashboard/app/models/lesson.rb:233]: how are the scripts which _are_ migrated rendering this markdown?

File: dashboard/app/models/lesson.rb [hunk 3]
```diff
@@ -406,7 +411,9 @@ def i18n_hash
       script.name => {
         'lessons' => {
           key => {
-            'name' => name
+            'name' => name,
+            'description_student' => student_overview,
+            'description_teacher' => overview
```
Comment (Hamms) [inline dashboard/app/models/lesson.rb:416]: please also update the comment for this method

File: dashboard/app/models/lesson.rb (hunk 2 above)
Comment (bethanyaconnor) [reply dashboard/app/models/lesson.rb:233]: Still using `SafeMarkdown` but I believe the markdown is imported from CB at some point. Not completely sure but I didn't want to change the behavior for old scripts.

File: dashboard/app/models/lesson.rb (hunk 1 above)
Comment (bethanyaconnor) [reply dashboard/app/models/lesson.rb]: I just copied over the old access. I'll fix it while I'm here though.

File: dashboard/app/models/lesson.rb (hunk 2 above)
Comment (bethanyaconnor) [reply dashboard/app/models/lesson.rb:233]: Sorry I got your question backwards! The `SafeMarkdown` is already being used so it just renders as any other markdown we have.

**Review by dmcavoy — APPROVED** [review]
Looks good to me! So excited we don't have to pull these from CB anymore

**Pull request merged**